portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -42,17 +42,17 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String defaultCommentLimit = "5";
-    int commentLimit = getIntInpOrDefault(request, "commentLimit", defaultCommentLimit);
+    int commentLimit = getPositiveIntInputOrDefault(request, "commentLimit", defaultCommentLimit);
 
     String defaultPageNumber = "0";
-    int pageNumber = getIntInpOrDefault(request, "pageNumber", defaultPageNumber);
-
-    String defaultSortMethod = "true";
-    String sortMethod = getParameterOrDefault(request, "sortMethod", defaultSortMethod);
+    int pageNumber = getIntInputOrDefault(request, "pageNumber", defaultPageNumber);
 
+    Boolean sortDescending = Boolean.parseBoolean
+                                    (getParameterOrDefault(request, "sortDirection", "true"));
```

Comment:
Interestingly the standard parseBoolean would not throw an exception to this! 
- parseBoolean(String) returns true if the String is (case-insensitive) "true", otherwise false

https://stackoverflow.com/questions/1762878/how-to-check-if-string-value-is-boolean-type-in-java. 

I've written a method as suggested and used BooleanUtils which returns null if it can't be parsed into some boolean, not just true...